### PR TITLE
multiple fixes of small things and QOL

### PR DESCRIPTION
**Pull request by liz3**
# multiple fixes of small things and QOL

### What's Changed:
* Fixes `mkdirAll` which would fail if one of the directories would already exist.
* adds `eprint` and` eprintln` to the IO module
* add a CMake option to **not** relocate the main executable(i use this to have two builds debug/not debug at the same time`
* Adds a git hash and debug info to the version
* fixes some stuff in the docs

#

### Type of Change:

- [x] Bug fix
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping:

- [x] Tests have been updated to reflect the changes done within this PR (if applicable).
- [x] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

**Comment by liz3** [comment]
@briandowns @Jason2605

**Review by briandowns — APPROVED** [review]
LGTM

File: tests/path/isDir.du [hunk 1]
```diff
@@ -3,7 +3,7 @@
  *
  * Testing Path.isDir()
  *
- * Returns true if the given string is a path to a directory, else false. (Linux only) 
+ * Returns true if the given string is a path to a directory, else false.`
```
Comment (liz3) [inline tests/path/isDir.du:6]:
```suggestion
 * Returns true if the given string is a path to a directory, else false.

```

**Comment by Jason2605** [comment]
Nice, thanks for this!

**Pull request merged**